src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/FlagAuthoritySystem.java

```diff
@@ -113,4 +122,58 @@ public void onGiveItemToCharacterHoldItem(GiveItemEvent event, EntityRef item, I
                 delayManager.cancelDelayedAction(item, LASUtils.DROPPED_FLAG);
         }
     }
+    /**
+     * Iterates through the inventory to locate the flag and then sends a {@link DropItemRequest} to remove the flag from
+     * the player's inventory.
+     *
+     * @param event The FlagDrop event sent to the player.
+     * @param targetPlayer The entity which should drop the flag.
+     */
+    @ReceiveEvent
+    public void dropFlagRequest(DropFlagEvent event, EntityRef targetPlayer) {
+        int inventorySize = inventoryManager.getNumSlots(targetPlayer);
+        for (int slotNumber = 0; slotNumber <= inventorySize; slotNumber++) {
+            EntityRef inventorySlot = inventoryManager.getItemInSlot(targetPlayer, slotNumber);
+            Optional<BlockItemComponent> inventoryItem = Optional.ofNullable(inventoryManager.getItemInSlot(targetPlayer,
+                    slotNumber).getComponent(BlockItemComponent.class));
+            inventoryItem.ifPresent(item -> {
+                if (item.blockFamily.getURI().toString().equals(event.getFlagTeam())) {
+                    flagSlot = inventorySlot;
+                }
+            });
+        }
+        Vector3fc startPosition = targetPlayer.getComponent(LocationComponent.class).getLocalPosition();
+        Vector3f impulse = event.getAttackingPlayer().getComponent(LocationComponent.class).getLocalPosition()
+                .add(startPosition, new Vector3f()).div(2f);
+        targetPlayer.send(new DropItemRequest(flagSlot, targetPlayer, impulse, startPosition));
+    }
+
+    /**
+     * Places the flag at the base and removes the flag from the players inventory.
+     *
+     * @param event The MoveFlagToBase event sent to the player.
+     * @param playerEntity The entity from which the flag should be removed.
```

Comment:
Ah, now I get why this event is sent against the player and not the flag ... :thinking: